### ISSUE TITLE
weird text in heatmap (also caused by a recent change)

### ISSUE THREAD

**Issue by jowens** [vega/vega-lite#735]
Here's a subset of a heatmap:

```
{"data": {"values": [{"alpha": 24, "beta": 24, "m_teps": 2289.174072265625, "m_teps_rounded": 2289.0, "dataset": "soc-LiveJournal1"}]}, "marktype": "text", "config": {"textCellWidth": 25}, "encoding": {"color": {"type": "Q", "name": "m_teps"}, "text": {"type": "Q", "name": "m_teps_rounded"}, "col": {"type": "O", "name": "beta"}, "row": {"type": "O", "name": "alpha"}}}
```

I want to display the ```m_teps``` field, but it is too long, so I rounded it and figure I'll display that instead (```m_teps_rounded```). But instead of displaying "2289" like it used to, now it displays 2.28900k, which in my view is kind of odd. 

It would be nice to have an auto-rounding feature so I don't have to round numbers myself for display, but in the absence of that, 2289 seems preferable to 2.28900k.

(I'm on master for both vega and vega-lite.)

<img width="345" alt="vega_scaffold" src="https://cloud.githubusercontent.com/assets/976970/11222224/30dfdcda-8d1e-11e5-8894-b67efd68b883.png">

**Comment by kanitw** [comment]
Just for experimenting, if you change `'s'` to `''` [in line 156 of `config.schema.ts`](https://github.com/vega/vega-lite/blob/master/src/schema/config.schema.ts#L156), does that work better for you?

Automatically determining number format is so hard, we're figuring out a good default.  Right now, we use `'s'` which is good for avoiding number collision for x-axis, but sometimes produce weird results like you are facing.

**Comment by jowens** [comment]
For me, yes, that appears to work better. 

I can imagine this is a challenging problem to generalize.
<img width="411" alt="vega_scaffold" src="https://cloud.githubusercontent.com/assets/976970/11222711/e17b2e76-8d20-11e5-91a9-c7db0906ef77.png">

**Comment by kanitw** [comment]
Yep. 

For now, please modify your spec to explicitly specify `format = ""`

```
"text": {
  "type": "Q",
  "name": "m_teps_rounded"
  "format": ""
}
```

And that should works fine for you.

**Comment by jowens** [comment]
Hate to say this, but that "format" fix doesn't appear to work.

<img width="757" alt="2____documents_working_gunrock-io_scripts__jowens_mission-burrito-2_and_vega_scaffold" src="https://cloud.githubusercontent.com/assets/976970/11223358/6869fd24-8d24-11e5-900d-82a7a7d021c4.png">

**Comment by kanitw** [comment]
Apologies.   I just realized there is a very subtle bug.   

I’ve just pushed a fix for you in master (See 99d421baa6f3dc7ad67a03b74bcc0dda2f313ff7 - it’s so subtle ….)

**Comment by jowens** [comment]
Ah, that's better. I confirm the fix works. Every time I see JS's "rules" for comparing things like "" and undefined and 0, I feel like gouging my eyes out with a fork.
